i have the same error, my gcc version is 9.5.0
```bash
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/local/libexec/gcc/x86_64-pc-linux-gnu/9.5.0/lto-wrapper
Target: x86_64-pc-linux-gnu
Configured with: ../configure -enable-checking=release -enable-languages=c,c++ -disable-multilib
Thread model: posix
gcc version 9.5.0 (GCC) 
```
error when i make
``` bash
I llama.cpp build info: 
I UNAME_S:  Linux
I UNAME_P:  x86_64
I UNAME_M:  x86_64
I CFLAGS:   -I.              -O3 -std=c11   -fPIC -DNDEBUG -Wall -Wextra -Wpedantic -Wcast-qual -Wdouble-promotion -Wshadow -Wstrict-prototypes -Wpointer-arith -pthread -march=native -mtune=native
I CXXFLAGS: -I. -I./examples -O3 -std=c++11 -fPIC -DNDEBUG -Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -Wno-multichar -pthread -march=native -mtune=native
I LDFLAGS:  
I CC:       cc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
I CXX:      g++ (GCC) 9.5.0

cc  -I.              -O3 -std=c11   -fPIC -DNDEBUG -Wall -Wextra -Wpedantic -Wcast-qual -Wdouble-promotion -Wshadow -Wstrict-prototypes -Wpointer-arith -pthread -march=native -mtune=native   -c ggml.c -o ggml.o
ggml.c: In function ‘bytes_from_nibbles_32’:
ggml.c:536:27: warning: implicit declaration of function ‘_mm256_set_m128i’; did you mean ‘_mm256_set_epi8’? [-Wimplicit-function-declaration]
     const __m256i bytes = _mm256_set_m128i(_mm_srli_epi16(tmp, 4), tmp);
                           ^~~~~~~~~~~~~~~~
                           _mm256_set_epi8
ggml.c:536:27: error: incompatible types when initializing type ‘__m256i {aka const __vector(4) long long int}’ using type ‘int’
ggml.c: In function ‘ggml_graph_export_leaf’:
ggml.c:14584:39: warning: format ‘%lld’ expects argument of type ‘long long int’, but argument 6 has type ‘int64_t {aka const long int}’ [-Wformat=]
     fprintf(fout, "%-6s %-12s %8d %8lld %8lld %8lld %8lld %16zu %16zu %16zu %16zu %16p %16s\n",
                                   ~~~~^
                                   %8ld
ggml.c:14588:13:
             ne[0], ne[1], ne[2], ne[3],
             ~~~~~                      
ggml.c:14584:45: warning: format ‘%lld’ expects argument of type ‘long long int’, but argument 7 has type ‘int64_t {aka const long int}’ [-Wformat=]
     fprintf(fout, "%-6s %-12s %8d %8lld %8lld %8lld %8lld %16zu %16zu %16zu %16zu %16p %16s\n",
                                         ~~~~^
                                         %8ld
ggml.c:14588:20:
             ne[0], ne[1], ne[2], ne[3],
                    ~~~~~                     
ggml.c:14584:51: warning: format ‘%lld’ expects argument of type ‘long long int’, but argument 8 has type ‘int64_t {aka const long int}’ [-Wformat=]
     fprintf(fout, "%-6s %-12s %8d %8lld %8lld %8lld %8lld %16zu %16zu %16zu %16zu %16p %16s\n",
                                               ~~~~^
                                               %8ld
ggml.c:14588:27:
             ne[0], ne[1], ne[2], ne[3],
                           ~~~~~                    
ggml.c:14584:57: warning: format ‘%lld’ expects argument of type ‘long long int’, but argument 9 has type ‘int64_t {aka const long int}’ [-Wformat=]
     fprintf(fout, "%-6s %-12s %8d %8lld %8lld %8lld %8lld %16zu %16zu %16zu %16zu %16p %16s\n",
                                                     ~~~~^
                                                     %8ld
ggml.c:14588:34:
             ne[0], ne[1], ne[2], ne[3],
                                  ~~~~~                   
ggml.c: In function ‘ggml_graph_export_node’:
ggml.c:14598:44: warning: format ‘%lld’ expects argument of type ‘long long int’, but argument 7 has type ‘int64_t {aka const long int}’ [-Wformat=]
     fprintf(fout, "%-6s %-6s %-12s %8d %8lld %8lld %8lld %8lld %16zu %16zu %16zu %16zu %8d %16p %16s\n",
ggml.c:14603:13:
             ne[0], ne[1], ne[2], ne[3],
             ~~~~~                           
ggml.c:14598:50: warning: format ‘%lld’ expects argument of type ‘long long int’, but argument 8 has type ‘int64_t {aka const long int}’ [-Wformat=]
     fprintf(fout, "%-6s %-6s %-12s %8d %8lld %8lld %8lld %8lld %16zu %16zu %16zu %16zu %8d %16p %16s\n",
                                              ~~~~^
                                              %8ld
ggml.c:14603:20:
             ne[0], ne[1], ne[2], ne[3],
                    ~~~~~                          
ggml.c:14598:56: warning: format ‘%lld’ expects argument of type ‘long long int’, but argument 9 has type ‘int64_t {aka const long int}’ [-Wformat=]
     fprintf(fout, "%-6s %-6s %-12s %8d %8lld %8lld %8lld %8lld %16zu %16zu %16zu %16zu %8d %16p %16s\n",
                                                    ~~~~^
                                                    %8ld
ggml.c:14603:27:
             ne[0], ne[1], ne[2], ne[3],
                           ~~~~~                         
ggml.c:14598:62: warning: format ‘%lld’ expects argument of type ‘long long int’, but argument 10 has type ‘int64_t {aka const long int}’ [-Wformat=]
     fprintf(fout, "%-6s %-6s %-12s %8d %8lld %8lld %8lld %8lld %16zu %16zu %16zu %16zu %8d %16p %16s\n",
                                                          ~~~~^
                                                          %8ld
ggml.c:14603:34:
             ne[0], ne[1], ne[2], ne[3],
                                  ~~~~~                        
ggml.c: In function ‘ggml_graph_export’:
ggml.c:14631:34: warning: format ‘%llu’ expects argument of type ‘long long unsigned int’, but argument 4 has type ‘uint64_t {aka long unsigned int}’ [-Wformat=]
         fprintf(fout, "%-16s %8llu\n", "eval",    size_eval);
                              ~~~~^
                              %8lu
ggml.c: In function ‘ggml_graph_import’:
ggml.c:14865:9: warning: ignoring return value of ‘fread’, declared with attribute warn_unused_result [-Wunused-result]
         fread(data->data, sizeof(char), fsize, fin);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Makefile:211: recipe for target 'ggml.o' failed
make: *** [ggml.o] Error 1
```
please help me.

--- a/ggml.c
+++ b/ggml.c
@@ -474,6 +474,8 @@ static const size_t CACHE_LINE_SIZE_F32 = CACHE_LINE_SIZE/sizeof(float);
 // quantization
 //
 
+#define MM256_SET_M128I(a, b) _mm256_insertf128_si256(_mm256_castsi128_si256(b), (a), 1)
+
 #if defined(__AVX__) || defined(__AVX2__) || defined(__AVX512F__) || defined(__SSSE3__)
 // multiply int8_t, add results pairwise twice
 static inline __m128i mul_sum_i8_pairs(const __m128i x, const __m128i y) {
@@ -533,7 +535,7 @@ static inline __m256i bytes_from_bits_32(const uint8_t * x) {
 static inline __m256i bytes_from_nibbles_32(const uint8_t * rsi)
 {
     const __m128i tmp = _mm_loadu_si128((const __m128i *)rsi);
-    const __m256i bytes = _mm256_set_m128i(_mm_srli_epi16(tmp, 4), tmp);
+    const __m256i bytes = MM256_SET_M128I(_mm_srli_epi16(tmp, 4), tmp);
     const __m256i lowMask = _mm256_set1_epi8( 0xF );
     return _mm256_and_si256(lowMask, bytes);
 }
@@ -606,7 +608,7 @@ static inline __m256i bytes_from_bits_32(const uint8_t * x) {
     bytesh = _mm_or_si128(bytesh, bit_mask);
     bytesl = _mm_cmpeq_epi8(bytesl, _mm_set1_epi64x(-1));
     bytesh = _mm_cmpeq_epi8(bytesh, _mm_set1_epi64x(-1));
-    return _mm256_set_m128i(bytesh, bytesl);
+    return MM256_SET_M128I(bytesh, bytesl);
 }
 
 // Unpack 32 4-bit fields into 32 bytes
@@ -619,7 +621,7 @@ static inline __m256i bytes_from_nibbles_32(const uint8_t * rsi)
     const __m128i lowMask = _mm_set1_epi8(0xF);
     tmpl = _mm_and_si128(lowMask, tmpl);
     tmph = _mm_and_si128(lowMask, tmph);
-    return _mm256_set_m128i(tmph, tmpl);
+    return MM256_SET_M128I(tmph, tmpl);
 }
 
 // add int16_t pairwise and return as float vector
@@ -627,7 +629,7 @@ static inline __m256 sum_i16_pairs_float(const __m128i xh, const __m128i xl) {
     const __m128i ones = _mm_set1_epi16(1);
     const __m128i summed_pairsl = _mm_madd_epi16(ones, xl);
     const __m128i summed_pairsh = _mm_madd_epi16(ones, xh);
-    const __m256i summed_pairs = _mm256_set_m128i(summed_pairsh, summed_pairsl);
+    const __m256i summed_pairs = MM256_SET_M128I(summed_pairsh, summed_pairsl);
     return _mm256_cvtepi32_ps(summed_pairs);
 }
 
@@ -2290,7 +2292,7 @@ static void ggml_vec_dot_q4_0_q8_0(const int n, float * restrict s, const void *
         const __m128i i32_1 = mul_sum_i8_pairs(bx, by);
 
         // Convert int32_t to float
-        __m256 p = _mm256_cvtepi32_ps(_mm256_set_m128i(i32_0, i32_1));
+        __m256 p = _mm256_cvtepi32_ps(MM256_SET_M128I(i32_0, i32_1));
 
         // Apply the scale, and accumulate
         acc = _mm256_add_ps(_mm256_mul_ps( d, p ), acc);
@@ -2766,7 +2768,7 @@ static void ggml_vec_dot_q5_0_q8_0(const int n, float * restrict s, const void *
         __m128i bxh = _mm256_extractf128_si256(bx, 1);
         bxl = _mm_or_si128(bxl, bxhil);
         bxh = _mm_or_si128(bxh, bxhih);
-        bx = _mm256_set_m128i(bxh, bxl);
+        bx = MM256_SET_M128I(bxh, bxl);
 
         const __m256i by = _mm256_loadu_si256((const __m256i *)y[i].qs);
 
@@ -3022,7 +3024,7 @@ static void ggml_vec_dot_q5_1_q8_1(const int n, float * restrict s, const void *
         __m128i bxh = _mm256_extractf128_si256(bx, 1);
         bxl = _mm_or_si128(bxl, bxhil);
         bxh = _mm_or_si128(bxh, bxhih);
-        bx = _mm256_set_m128i(bxh, bxl);
+        bx = MM256_SET_M128I(bxh, bxl);
 
         const __m256 dy = _mm256_set1_ps(y[i].d);
         const __m256i by = _mm256_loadu_si256((const __m256i *)y[i].qs);

